integration-tests/smoke/ccip/ccip_rmn_test.go

```diff
@@ -653,21 +654,32 @@ func (tc rmnTestCase) callContractsToCurseAndRevokeCurse(ctx context.Context, t
 		}
 
 		for subjectDescription, revokeAfter := range cursedSubjects {
-			subj := reader.GlobalCurseSubject
-			if subjectDescription != globalCurse {
-				subj = chainSelectorToBytes16(tc.pf.chainSelectors[subjectDescription])
+			curseActions := make([]changeset.CurseAction, 0)
+
+			if subjectDescription == globalCurse {
+				curseActions = append(curseActions, changeset.CurseChain(remoteSel))
+			} else {
+				curseActions = append(curseActions, changeset.CurseLane(remoteSel, (tc.pf.chainSelectors[subjectDescription])))
 			}
-			t.Logf("cursing subject %d (%d)", subj, subjectDescription)
-			txCurse, errCurse := chState.RMNRemote.Curse(chain.DeployerKey, subj)
-			_, errConfirm := deployment.ConfirmIfNoError(chain, txCurse, errCurse)
-			require.NoError(t, errConfirm)
+
+			_, err := changeset.NewRMNCurseChangeset(envWithRMN.Env, changeset.RMNCurseConfig{
+				CurseActions: curseActions,
+				Reason:       "test curse",
+			})
+			t.Error(err)
 
 			go func() {
 				<-time.NewTimer(revokeAfter).C
-				t.Logf("revoking curse on subject %d (%d)", subj, subjectDescription)
-				txUncurse, errUncurse := chState.RMNRemote.Uncurse(chain.DeployerKey, subj)
-				_, errConfirm = deployment.ConfirmIfNoError(chain, txUncurse, errUncurse)
-				require.NoError(t, errConfirm)
+				t.Logf("revoking curse on subject %d (%d)", subjectDescription, subjectDescription)
+
+				_, err := changeset.NewRMNUncurseChangeset(envWithRMN.Env, changeset.RMNCurseConfig{
+					CurseActions: curseActions,
+					Reason:       "test uncurse",
+				})
+				require.NoError(t, err) //nolint
```

Comment:
you can use error group here, if error occurs here the outer test will not stop. it's better to have error group and then check the error outside of the go routine